csharp/src/Drivers/Apache/Spark/SparkConnection.cs

```diff
@@ -269,32 +271,58 @@ protected override async ValueTask<TProtocol> CreateProtocolAsync()
                 Trace.TraceError($"key = {property} value = {properties[property]}");
             }
 
-            string hostName = properties[SparkParameters.HostName];
-            string path = properties[SparkParameters.Path];
-            string token;
-
-            if (properties.ContainsKey(SparkParameters.Token))
-                token = properties[SparkParameters.Token];
-            else
-                token = properties[SparkParameters.Password];
-
-            HttpClient httpClient = new HttpClient();
-            httpClient.BaseAddress = new UriBuilder(Uri.UriSchemeHttps, hostName, -1, path).Uri;
-            httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+            properties.TryGetValue(SparkParameters.HostName, out string? hostName);
+            properties.TryGetValue(SparkParameters.Scheme, out string? scheme);
```

Comment:
What's the rationale behind accepting a scheme as a separate parameter versus e.g. an url parameter? It seems to increase the chance of unexpected parameter combinations?

Reply:
"uri" is even a standard ADBC option name.

Reply:
Added support for "uri" and removed "scheme". Also reference `AdbcOptions.Username/Password/Uri`